mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/MongoDBDialect.java

```diff
@@ -1333,12 +1333,16 @@ else if ( operation instanceof InsertOrUpdateAssociationOperation ) {
 				if ( storageStrategy == AssociationStorageStrategy.IN_ENTITY ) {
 					writeConcern = mergeWriteConcern( writeConcern, getWriteConcern( associationContext ) );
 					if ( insertStatement != null ) {
+						// The association is updated in a new document
 						MongoHelpers.setValue( insertStatement, collectionRole, rows );
 					}
 					else {
-						MongoHelpers.setValue( ( (MongoDBTupleSnapshot) associationContext.getEntityTuplePointer().getTuple().getSnapshot() ).getDbObject(),
-								associationKey.getMetadata().getCollectionRole(), toStore );
-						addSetToQuery( updateStatement, collectionRole, toStore );
+						// The association is updated on an existing document
+						Document document = getDocument( association, associationContext );
+						if ( document != null ) {
```

Comment:
Do you know when this situation happens? Don't we lose some data in the case the document is null?